nix-darwin/default.nix

```diff
@@ -77,7 +96,10 @@ in
     system.activationScripts.postActivation.text =
       concatStringsSep "\n" (mapAttrsToList (username: usercfg: ''
         echo Activating home-manager configuration for ${username}
-        sudo -u ${username} -i ${usercfg.home.activationPackage}/activate
+        sudo -u ${username} -i ${pkgs.writeShellScript "activation-${username}" ''
+          export HOME_MANAGER_BACKUP_EXT='${lib.escapeShellArg cfg.backupFileExtension}'
```

Comment:
Does this work if `backupFileExtension == null`?

Reply:
Good point, this should be behind a `lib.optionalString`.

Reply:
Fixed